src/application/services/useTools.ts

```diff
@@ -0,0 +1,82 @@
+import { type Ref, ref, watch } from 'vue';
+import { useAppState } from './useAppState';
+import type EditorTool from '@/domain/entities/EditorTool';
+import { loadScript } from '@/infrastructure/utils/load-script';
+import type { EditorConfig } from '@editorjs/editorjs';
+
+/**
+ * Downloaded tools data structure
+ */
+type DownloadedTools = EditorConfig['tools'];
+
+/**
+ * Service for load user tools
+ *
+ * @param noteTools - note tools
+ */
+export function useTools(noteTools: Ref<EditorTool[]>): {
+  tools: Ref<DownloadedTools | undefined>;
+} {
+  /**
+   * User notes tools
+   */
+  const { userEditorTools } = useAppState();
+  const tools = ref<DownloadedTools | undefined>();
+
+  /**
+   * Download all the user tools and return a map
+   *
+   * @param toolsList - tools data
+   */
+  async function downloadTools(toolsList: EditorTool[]): Promise<DownloadedTools> {
```

Comment:
this logic seems to be in an `infrastructure` layer since we will need a store for loaded tools to prevent multiple loading.